tests/test_run.py

```diff
@@ -139,3 +147,312 @@ def test_train_freq_fail(train_freq):
             train_freq=train_freq,
         )
         model.learn(total_timesteps=250)
+
+
+def test_dqn_train_with_batch_norm():
+    model = DQN(
+        "MlpPolicy",
+        "CartPole-v1",
+        policy_kwargs=dict(net_arch=[16, 16], features_extractor_class=FlattenBatchNormDropoutExtractor),
+        learning_starts=0,
+        tau=0,
+        seed=1,
+    )
+
+    (
+        q_net_bias_before,
+        q_net_running_mean_before,
+        q_net_target_bias_before,
+        q_net_target_running_mean_before,
+    ) = clone_dqn_batch_norm_stats(model)
+
+    model.learn(total_timesteps=200)
+
+    (
+        q_net_bias_after,
+        q_net_running_mean_after,
+        q_net_target_bias_after,
+        q_net_target_running_mean_after,
+    ) = clone_dqn_batch_norm_stats(model)
+
+    assert ~th.isclose(q_net_bias_before, q_net_bias_after).all()
+    assert ~th.isclose(q_net_running_mean_before, q_net_running_mean_after).all()
+
+    assert th.isclose(q_net_target_bias_before, q_net_target_bias_after).all()
+    assert th.isclose(q_net_target_running_mean_before, q_net_target_running_mean_after).all()
+
+
+def test_dqn_collect_rollouts_with_batch_norm():
+    model = DQN(
+        "MlpPolicy",
+        "CartPole-v1",
+        policy_kwargs=dict(net_arch=[16, 16], features_extractor_class=FlattenBatchNormDropoutExtractor),
+        seed=1,
+    )
+
+    (
+        q_net_bias_before,
+        q_net_running_mean_before,
+        q_net_target_bias_before,
+        q_net_target_running_mean_before,
+    ) = clone_dqn_batch_norm_stats(model)
+
+    total_timesteps, callback = model._setup_learn(total_timesteps=100, eval_env=model.get_env())
```

Comment:
why not calling `.learn()` directly?

Reply:
Do you mean call `.learn()` with `learning_starts >= total_timesteps`? I thought about this, but the on-policy algorithms don't have a `learning_starts` parameter, so I don't think it would work for them. 

I can change the tests for the off-policy algorithms if you prefer?

Reply:
you can do as we do for other tests, where we change `learning_starts` to zero if it is an off-policy algorithm, for instance https://github.com/DLR-RM/stable-baselines3/blob/master/tests/test_save_load.py#L178

Reply:
Sorry, I'm not sure I understand. The reason for this test is to check that the batch norm stats are _not_ updated when `.collect_rollouts()` is called. If I set `learning_starts = 0` before calling `.learn()`, then `.train()` will be called after each call to `.collect_rollouts()`, which will cause the batch norm stats to be updated and the test will fail.

Reply:
> The reason for this test is to check that the batch norm stats are not updated when .collect_rollouts() is called. 


I see, then please do a first call to `learn()` with `num_timesteps < learning_starts` and that should do the trick, no?

Reply:
Yes, that works. I've made that change for the off-policy algorithms.